neps/nep-0330.md

```diff
@@ -31,23 +35,36 @@ There is a lot of information that can be held about a contract. Ultimately, we
 
 Successful implementations of this standard will introduce a new  (`ContractSourceMetadata`) struct that will hold all the necessary information to be queried for. This struct will be kept on the contract level.
 
-The metadata will include three optional fields:
+The metadata will include optional fields:
 
-- `version`: a string that references the specific commit hash or version of the code that is currently deployed on-chain. This can be included regardless of whether or not the contract is open-sourced and can also be used for organizational purposes.
+- `version`: a string that references the specific commit hash or version of the code currently deployed on-chain. If the link is to a GitHub repo URL, this is crucial, as it is needed to make the contract reproducible.
 - `link`: a string that references the link to the open-source code. This can be anything such as Github or a CID to somewhere on IPFS.
```

Comment:
@lachlanglen Would this approach of a "frozen" link work for you? I believe it should be a good replacement for a `commit_hash` field which assumes that the code is stored on a public git provider.

Reply:
> Thus, I would move this build-related source code link to the `build_details` and don't touch the top-level `link` or `version` at all.

I agree. I appreciate the approach of using a "frozen" link, as it appears to resolve all the issues related to the "link" and "version" fields. The reason I didn't add it initially was to avoid confusion for anyone reviewing the metadata. Therefore, the name of the new field should be distinctly different from "link". In my opinion, "source_snapshot" fits well.

Reply:
```ts
type BuildInfo {
    build_environment_ref: string, // reference to a reproducible build environment docker image, e.g. "sourcescan/cargo-near@sha256:bf488476d9c4e49e36862bbdef2c595f88d34a295fd551cc65dc291553849471" or something else pointing to the build environment.
    source_code_snapshot: string, // reference to the source code snapshot that was used to build the contract, e.g. "git+https://github.com/near-DevHub/neardevhub-contract.git#335e89edec95d56a4744e7160c3fd590e41ec38e" or "ipfs://<ipfs-hash>"
    contract_path: string|null, // relative path to contract folder within the source code e.g. "src/contract". Often, it is the root of the repository, so can be omitted. 
    build_command: string[], // the exact command that was used to build the contract, with all the flags e.g. ["cargo", "near", "build", "no-abi"].
}
```

Now using `source_code_snapshot` in 18968ea09c965845b92ab7fdb39a8a1df319b7cd